### PR TITLE
#171380740 integrate coveralls

### PR DESCRIPTION
**Pull request by Ugizwenayo-Divine**
#### What does this PR do?
- integrate coveralls code coverage service

#### Description of Task to be completed?
- install coveralls 
- add stackup2-barefoot repo on coveralls
- create .coveralls.yml file
- add coveralls badge in readme file
- write coveralls script which runs coverage
- configure nyc for generating a report

#### How should this be manually tested?
- After the tests are written, it should give you the coverage of the tested lines.
- you need to add this command "nyc --reporter=html --reporter=text" 
in the test script so that it generates the coverage.
#### Any background context you want to provide?
N/A
#### Question
waiting for the unit test PR to be merged

#### What are the relevant pivotal tracker stories?
 [#171380740] (https://www.pivotaltracker.com/story/show/171380740)

### CODE REVIEW

**Review by ricahcyuzuzo — COMMENTED** [review]
Hey Divine, Where is the badge of coveralls in your readme file? I'd like to request a badge in a readme file

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
### My Suggestions

* I suggest you to change the `in-progress` badge into `WIP` on this PR
* I suggest also to install `nyc` package which calls npm scripts to run tests and report the coverage
* I suggest you to add scripts in `package.json` to run coverage,

**Comment by Ugizwenayo-Divine** [comment]
> Hey Divine, Where is the badge of coveralls in your readme file? I'd like to request a badge in a readme file
I have added the badge in readme 
thank you

**Review by descholar-ceo — COMMENTED** [review]
Cool, I see that you tried to implement all of the suggestions which is good, but I pulled your branch, and the one which has a sample test, unfortinately, I am no seeing any coverage.

Could you try to see work on it and figure out what is a problem?

**Review by descholar-ceo — DISMISSED** [review]

**Review by since9629 — APPROVED** [review]
Great job @Ugizwenayo-Divine

**Review by descholar-ceo — APPROVED** [review]
LGTM

File: .coveralls.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+repo_token: IDIbRhOhSENsSUw1oXoka4q0WbtjJZWcs
```
Comment (Muhire-Josue) [inline .coveralls.yml]: I think you need to fix this next line issue

**Review by Muhire-Josue — APPROVED** [review]
Great work Divine. I left a little comment.

**Review by muchristian — APPROVED** [review]
good job @Ugizwenayo-Divine

**Review by Muhire-Josue — APPROVED** [review]
Looks good to me.

**Review by mifeille — APPROVED** [review]

**Pull request merged**